### PR TITLE
Implemented SpikeGLX extractor for neuropixels

### PR DESCRIPTION
**Pull request by alejoe91**
- read number of channels
- loads channel locations (assuming standard configuration)

### CODE REVIEW

**Comment by colehurwitz31** [comment]
Looks great! Is it possible to memory map the time series? I was wondering if we should do that for the BinDatExtractor also... if it is possible.

**Comment by alejoe91** [comment]
I think it is memmaped. Check out the _read_binary function

**Comment by colehurwitz31** [comment]
I see! That is great. However, we now have to read_binary functions (in SpikeGLX and BinDat). Is that repeated code that we can move to tools.py?

**Comment by alejoe91** [comment]
Yes that's exactly the same, will do it

**Comment by alejoe91** [comment]
done :)

File: spikeextractors/extractors/spikeglxextractor/spikeglxrecordingextractor.py [hunk 1]
```diff
@@ -97,6 +96,9 @@ def _read_binary(file, numchan, dtype, frames_first, offset):
 def _parse_spikeglx_metafile(metafile):
     tot_channels = None
     ap_channels = None
+    x_pitch = 21
+    y_pitch = 20
```
Comment (colehurwitz31) [inline spikeextractors/extractors/spikeglxextractor/spikeglxrecordingextractor.py]: Will pitch always be the same here?

Comment (colehurwitz31) [reply spikeextractors/extractors/spikeglxextractor/spikeglxrecordingextractor.py]: Is this only for Neuropixels?

**Comment by alejoe91** [comment]
yes for the neuropixels probe, but I could add optional args

**Comment by colehurwitz31** [comment]
If it is possible, we want the SpikeGLX extractor for any probe geometry, right? I may be misunderstanding what SpikeGLX encapsulates.

**Comment by alejoe91** [comment]
SpikeGLX only works with Neuropixels

**Comment by colehurwitz31** [comment]
Got it. So the num channels will always be 385? Or could they use a portion of the channels?

**Comment by alejoe91** [comment]
no it depends on the sync channels. It gets it from the metadata

**Comment by colehurwitz31** [comment]
Ok, so I guess the only change I would recommend is fixing the title of y_pitch in the gui params. Then I will merge. Awesome work Alessio!

**Pull request merged**